src/reduced_basis/rb_construction.C

```diff
@@ -682,6 +693,18 @@ void RBConstruction::add_scaled_matrix_and_vector(Number scalar,
             }
         }
     }
+  } // libmesh_try
+  libmesh_catch(...)
```

Comment:
Would it make sense to have a preceding catch block for `std::exception & e`, so you can preserve `e.what()`?  We don't have any great examples of that in libMesh (maybe one in rb_data_serialization.C) but it's standard in MOOSE and seems to be really helpful for diagnosis.

Reply:
I think that's a potential improvement, but we'd have to communicate the `e.what()` string(s) from all processors and combine or uniquify them in some way for it to be sensible to the user. And I would not actually be able to test this easily, since our own internal exception type currently does not derive from std::exception.